File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int p = 1;
        bool containtZ = false;
        int numZ = 0;
        vector<int> result;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                containtZ = true;
                numZ += 1;
            } else {
                p = p * nums[i];
            }
        }
        for (int i = 0; i < nums.size(); i++) {
            if (numZ > 1) {
                result.push_back(0);
                continue;
            }
            if (containtZ) {
                if (nums[i] == 0) {
                    result.push_back(p);
                } else {
                    result.push_back(0);
                }
            } else {
                result.push_back(p / nums[i]);
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)

        l = 0
        for num in s:
            if num - 1 not in s:
                v = 1
                j = num
                while j + 1 in s:    
                    j = j + 1
                    v = v + 1
                if v > l:
                    l = v
            
                l = max(l, v)
        return l